### ISSUE TITLE
refactor dapple install

### ISSUE THREAD

**Issue by mhhf** [nexusdev/dapple#190]
refactor install and dependencies: work with pure functions and clear cli type install args. dapphub as default
```
dapple install dappsys # install from dapphub
dapple install ipfs://asdasdas --ipfs
dapple install git://adasdasd --git
```

**Comment by mhhf** [comment]
When I think about this, we should probably deprecate it all together and make installing from dapphub the only option when installing from cli. 
@ryepdx @nmushegian Thoughts?

**Comment by nmushegian** [comment]
I'd like to support referring to packages by ipfs hash.

**Comment by ryepdx** [comment]
I feel like I can see both sides to this one, since removing the extra sources could simplify the codebase and since one could argue including additional installation sources might be an example of YAGNI. On the other hand, that sort of functionality is available in package managers such as NPM and Pip, so developers are probably going to approach Dapple with the expectation that it too will support installing from git repositories, HTTP servers, IPFS, etc. I'm inclined to follow the herd on this one because of that, but I'd also be interested in hearing counterarguments.

**Comment by dbrock** [comment]
Isn't "Install from IPFS hash" simply one of two steps in "Install from Dapphub" (the first step being "Download IPFS hash from Dapphub")?

**Comment by ryepdx** [comment]
Yeah, it is. Enabling installation from IPFS hash would basically just be a matter of changing the place from where the hash is pulled.
